### ISSUE TITLE
Document CSS/Tailwind autocomplete configuration for vscode and other editors

### ISSUE THREAD

**Issue by kang-sw** [DioxusLabs/dioxus#1493]
TL;DR: Install [this extension](https://marketplace.visualstudio.com/items?itemName=kang-sw.html-css-class-completion-everywhere) and add the following to your VSCode settings file:

    "html-css-class-completion.CustomizedLanguages": {
        "rust": ["class\\s*[:=]\\s*\"([\\w- ]*$)"]
    }

After that, when you type `class: "` in any context within your Rust source code, autocompletion suggestions will pop up based on your CSS class collection (from workspace css/html files).

![image](https://github.com/DioxusLabs/dioxus/assets/32210953/86befd7e-d004-4e96-b6fb-3d29cf8c816c)

---

Some of you might be familiar with the [IntelliSense for CSS class names in HTML](https://marketplace.visualstudio.com/items?itemName=Zignd.html-css-class-completion) VSCode extension. For those who aren't, this extension gathers CSS class names defined in the local file tree and the html `<style>` section to offer autocompletion. I found it extremely helpful when developing with React.

While developing with Dioxus, I felt the lack of this extension's capabilities. Upon inspecting the source code, I realized all features for autocompletion in Rust were already defined. The only missing piece was its connection to the configuration file. After writing a small bridge code and testing it, everything worked flawlessly.

I wanted to share this so others can easily utilize it as well.

---

For Tailwind, it seems to cache and autocomplete for already generated local CSS. However, it inherently doesn't for content that isn't written in CSS (since Tailwind deduces required CSS from user codes and generates new CSS). But, there's already an official guide on [how to integrate dioxus and tailwind cli](https://dioxuslabs.com/learn/0.4/cookbook/tailwind). You can refer to that for more details.

> Another idea that just popped into my mind: By placing the full version of Tailwind's CSS file in the local file tree, it might achieve the same effect.

---

> I'd prefer to have the feature integrated into the original extension rather than creating a fork, especially given the minor code adjustments. Unfortunately, it seems the original repository hasn't been active since 2021.

**Comment by ealmloff** [comment]
This would be great to add to a styling page similar to the current page in the cook book for tailwind, but for vanilla CSS!

**Comment by erayerdin** [comment]
This does not work anymore.

**Edit:** I've found [this](https://stackoverflow.com/a/75413114/2926992).
